src/test/java/org/opentripplanner/updater/trip/TimetableSnapshotSourceTest.java

```diff
@@ -68,28 +74,17 @@ public void setUp() {
     transitModel = model.transitModel();
 
     feedId = transitModel.getFeedIds().stream().findFirst().get();
-
-    final TripDescriptor.Builder tripDescriptorBuilder = TripDescriptor.newBuilder();
-
-    tripDescriptorBuilder.setTripId("1.1");
-    tripDescriptorBuilder.setScheduleRelationship(TripDescriptor.ScheduleRelationship.CANCELED);
-
-    final TripUpdate.Builder tripUpdateBuilder = TripUpdate.newBuilder();
-
-    tripUpdateBuilder.setTrip(tripDescriptorBuilder);
-
-    cancellation = tripUpdateBuilder.build().toByteArray();
   }
 
   @Test
-  public void testGetSnapshot() throws InvalidProtocolBufferException {
+  public void testGetSnapshot() {
     var updater = defaultUpdater();
 
     updater.applyTripUpdates(
       TRIP_MATCHER_NOOP,
       REQUIRED_NO_DATA,
-      fullDataset,
-      List.of(TripUpdate.parseFrom(cancellation)),
+      FULL_DATASET,
+      List.of(CANCELLATION),
```

Comment:
Much nicer! Why was this encoded and parsed in the original code?

Reply:
This test is very old. It possible to trace it back to 2012. My guess is that once upon a time, the class under test operated on binary messages, rather than parsed ones. Once the parsing was moved out of it, the test was just quickly patched up rather than properly rewritten.